### ISSUE TITLE
Infinite loop on loading training dataset

### ISSUE THREAD

**Issue by ziorufus** [OpenNMT/OpenNMT-py#1079]
Hello, I downloaded the 0.6.0 version of OpenNMT and I encountered a bug: the dataset during the training phase is loaded an infinite number of times.

```
...
[2018-11-28 17:48:30,940 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:49:15,827 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:50:00,966 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:50:46,060 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:51:31,272 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:52:16,291 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:53:00,940 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:53:46,221 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:54:31,436 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
[2018-11-28 17:55:16,504 INFO] Loading train dataset from /data/out-models/reverse-original-5/data.train.0.pt, number of examples: 146060
...
```

I think it's not normal. What can I do? Thanks!

**Comment by vince62s** [comment]
can you post your command line please ?

**Comment by eduamf** [comment]
Could you send the parameters used. Are you using one GPU?

**Comment by ziorufus** [comment]
```
/data/opennmt-py-virenv-backup/bin/python train.py -data /data/out-models/reverse-original-5/data -copy_attn -share_embeddings -reuse_copy_attn -save_model /data/out-models/reverse-original-5/ -gpu_ranks 2 -rnn_size 250 -word_vec_size 300 -seed 57
```

**Comment by shhshn** [comment]
You can use `CUDA_VISIBLE_DEVICES=2` and `-gpu_ranks 0` instead of using `-gpu_ranks 2`.

**Comment by ziorufus** [comment]
@shhshn it seems that the problem has disappeared, thank you very much!

**Comment by kangzhonghua** [comment]
Dear all
I am also encounter loop on loading training dataset.
---------ON 1 GPU---------------------------------
CUDA_VISIBLE_DEVICES=0 python -u train.py -data data/LCSTS2.1/LCSTS2.1 -save_model models/LCSTS2.1 -layers 6 -rnn_size 512 -word_vec_size 512 -transformer_ff 2048 -heads 8 -encoder_type transformer -decoder_type transformer -position_encoding -train_steps 200000 -max_generator_batches 2 -dropout 0.1 -batch_size 4096 -batch_type tokens -normalization tokens -accum_count 2 -optim adam -adam_beta2 0.998 -decay_method noam -warmup_steps 8000 -learning_rate 2 -max_grad_norm 0 -param_init 0 -param_init_glorot -label_smoothing 0.1 -valid_steps 10000 -save_checkpoint_steps 10000 -world_size 1 -gpu_ranks 0

[2019-03-13 11:17:27,525 INFO] encoder: 44516352
[2019-03-13 11:17:27,525 INFO] decoder: 76479316
[2019-03-13 11:17:27,525 INFO] * number of parameters: 120995668
[2019-03-13 11:17:27,530 INFO] Starting training on GPU: [0]
[2019-03-13 11:17:27,530 INFO] Start training loop and validate every 10000 steps...
[2019-03-13 11:17:28,352 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:17:54,672 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:18:03,610 INFO] Step 50/200000; acc:  23.77; ppl: 8261.54; xent: 9.02; lr: 0.00001; 3745/7472 tok/s;     36 sec

------------------ON 2 GPU---------------------------------------------------------
CUDA_VISIBLE_DEVICES=0,1 python -u train.py -data data/LCSTS2.1/LCSTS2.1 -save_model models/LCSTS2.1 -layers 6 -rnn_size 512 -word_vec_size 512 -transformer_ff 2048 -heads 8 -encoder_type transformer -decoder_type transformer -position_encoding -train_steps 200000 -max_generator_batches 2 -dropout 0.1 -batch_size 4096 -batch_type tokens -normalization tokens -accum_count 2 -optim adam -adam_beta2 0.998 -decay_method noam -warmup_steps 8000 -learning_rate 2 -max_grad_norm 0 -param_init 0 -param_init_glorot -label_smoothing 0.1 -valid_steps 10000-save_checkpoint_steps 10000 -world_size 2 -gpu_ranks 0 1

[2019-03-13 11:07:13,620 INFO] encoder: 44516352
[2019-03-13 11:07:13,620 INFO] decoder: 76479316
[2019-03-13 11:07:13,620 INFO] * number of parameters: 120995668
[2019-03-13 11:07:13,624 INFO] Starting training on GPU: [0, 1]
[2019-03-13 11:07:13,624 INFO] Start training loop and validate every 10000 steps...
[2019-03-13 11:07:14,433 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:07:30,584 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:07:47,083 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:07:57,869 INFO] Step 50/200000; acc:  24.14; ppl: 8678.35; xent: 9.07; lr: 0.00001; 6170/12357 tok/s;     44 sec
[2019-03-13 11:08:02,763 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:08:19,420 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:08:35,281 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:08:42,153 INFO] Step 100/200000; acc:  49.93; ppl: 3184.77; xent: 8.07; lr: 0.00001; 6165/12346 tok/s;     89 sec
[2019-03-13 11:08:51,936 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:09:07,928 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:09:24,591 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:09:26,763 INFO] Step 150/200000; acc:  49.92; ppl: 1871.43; xent: 7.53; lr: 0.00002; 6120/12259 tok/s;    133 sec
[2019-03-13 11:09:40,602 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:09:57,495 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:10:10,041 INFO] Step 200/200000; acc:  49.92; ppl: 1112.75; xent: 7.01; lr: 0.00002; 6308/12637 tok/s;    176 sec
[2019-03-13 11:10:13,303 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:10:30,022 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:10:45,993 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:10:54,385 INFO] Step 250/200000; acc:  57.70; ppl: 501.39; xent: 6.22; lr: 0.00003; 6157/12326 tok/s;    221 sec
[2019-03-13 11:11:02,506 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:11:18,572 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:11:35,038 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:11:38,754 INFO] Step 300/200000; acc:  73.55; ppl: 187.26; xent: 5.23; lr: 0.00004; 6153/12326 tok/s;    265 sec
[2019-03-13 11:11:51,019 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
[2019-03-13 11:12:07,792 INFO] Loading dataset from data/LCSTS2.1/LCSTS2.1.train.0.pt, number of examples: 100000
